src/plonesocial/microblog/browser/tiles/newpostbox.py

```diff
@@ -40,6 +40,8 @@ def post_context(self):
         ''' The context of this microblog post
         (the portal, a workspace, and so on...)
         '''
+        if self.thread_id:
+            return self.post_container.get(self.thread_id)
         return get_microblog_context(self.context)
 
```

Comment:
This looks invalid. If you have a thread_id it returns an object of type IStatusUpdate (a tweet). If you don't have a thread id it returns an IMicroblogContext (workspace) or ISiteRoot.

One cannot assume that IStatusUpdate us usable as a "normal" context without doing the extensive analysis and refactoring we have planned for the coming weeks.